babel.config.js

```diff
@@ -1,89 +1,152 @@
+ *
+ * // out
+ * import _ObjectFromEntries from "object-polyfills/from-entries";
+ * let obj = _ObjectFromEntries(entries);
+ * ```
+ *
+ * The "usage" part means Babel only polyfills what we use, not everything it possibly can.
+ * See https://github.com/babel/babel-polyfills/blob/main/docs/usage.md#method.
+ *
+ * ## shippedProposals
+ *
+ * Some tc39 proposals have shipped in browsers, but haven't been included in the spec yet.
+ * (Because it hasn't been published). Setting `shippedProposals` to true opts us into them.
+ *
+ * Note that we have to set shippedProposals in two places:
+ *
+ * - `@babel/preset-env` for syntax proposals (numeric separator)
+ * - `babel-plugin-polyfill-corejs3` for method proposals (String.prototype.matchAll)
+ */
 
+// This's used in the `babel-plugin-polyfill-corejs3` plugin below.
+// It's recommended to specify the version of core-js used up to the minor version, like "3.6".
+// Just specifying the major version won't inject modules added in minor releases.
+// See https://github.com/zloirock/core-js/blob/master/README.md#babelpreset-env.
+const path = require('path')
 const packageJSON = require(path.join(__dirname, 'package.json'))
-
-// RedwoodJS targets Node.js 12.x because this is the default version
-// for Netlify's functions.
-const TARGETS_NODE = '12.16'
-
-// Run `npx browserslist "defaults, not IE 11, not IE_Mob 11"` to see a list
-// of target browsers.
-const TARGETS_BROWSERS = ['defaults', 'not IE 11', 'not IE_Mob 11']
-
-// Warning! Recommended to specify used minor core-js version, like corejs: '3.6',
-// instead of corejs: '3', since with '3' it will not be injected modules
-// which were added in minor core-js releases.
-// https://github.com/zloirock/core-js/blob/master/README.md#babelpreset-env
-const CORE_JS_VERSION = packageJSON.devDependencies['core-js']
+const CORE_JS_VERSION = packageJSON.devDependencies['core-js-pure']
   .split('.')
   .slice(0, 2)
-  .join('.') // Produces: 3.12, instead of 3.12.1
-
-// We use the recommended babel configuration for monorepos, which is a base directory
-// `babel.config.js` file, but then use a per-project `.babelrc.js` file.
-// Learn more: https://babeljs.io/docs/en/config-files#monorepos
+  .join('.')
 
+// FIXME: this interface seems a little outdated.
 /** @type {import('@babel/core').TransformOptions} */
 module.exports = {
+  targets: {
+    node: '14.20',
+  },
+
+  assumptions: {
+    enumerableModuleMeta: true,
+  },
+
   presets: [
     [
       '@babel/preset-env',
       {
-        targets: { node: TARGETS_NODE },
-        useBuiltIns: 'usage',
-        corejs: {
-          version: CORE_JS_VERSION,
-          // List of supported proposals: https://github.com/zloirock/core-js/blob/master/docs/2019-03-19-core-js-3-babel-and-a-look-into-the-future.md#ecmascript-proposals
-          proposals: true,
-        },
+        // See https://babeljs.io/docs/en/babel-preset-env#shippedproposals.
+        shippedProposals: true,
```

Comment:
I still have to scan our packages to see if we were using a lot of proposals, but I don't think we are. Regardless, we have to make a decision if we want to use them because it opens the door to a lot of Stage 0-2 features that could change/get dropped. `shippedProposals` is a nice compromise because it allows using proposals that are Stage 4+. And we could always just include the proposals we want to use in an explicit `includes` array, so it's not all or nothing.